Revise docs for Auto-packets and auto-components (#2006)
o3de/o3de.org: content/docs/user-guide/gems/reference/multiplayer/multiplayer-gem/overview.md

{{< important >}}
Network components are the **only** built-in way to communicate between entities in O3DE. Due to the engine's ability to distribute and interact with entities across multiple hosts, there is no guarantee that any two entities have authoritative representations on the same host.
{{< /important >}}

## Network properties

The O3DE Multiplayer Gem uses *delta replication* to transmit property changes across hosts. The delta replication method uses a system where a server maintains information about both the last acknowledged state of each client and the current world state, and then sends the differences between the two states across the network. When the client receives this information, it updates the world state if the packet was received in the correct sequence.

> One important consideration of this style of replication is that it requires the client to send an acknowledgement (`ACK`) message back to the server. The O3DE AzNetworking framework offers reliable UDP packets, including acknowledgement.

Author: ```suggestion
One important consideration of this style of replication is that it requires the client to send an acknowledgement (`ACK`) message back to the server. The O3DE AzNetworking framework offers reliable UDP packets, including acknowledgement. ACK vector replication allows the AzNetworking layer to be fully aware of every packet that was received or dropped. 
```

Without taking advanced networking features into consideration, a server checks for a delta on every frame by:

1. Inspecting a snapshot of the world state against each client's acknowledged state.
1. If the world state is different from the last state that resulted in an `ACK`, generate a delta for that client.
1. Transmit the delta to the client.

Then, clients update themselves from a delta by:

1. Checking if the received delta is in sequence.
1. Extracting the delta information and updating the world state.

> 1. Sending an `ACK` to the server indicating the world was updated.

Author: ```suggestion
1. Sending an `ACK` to the server, indicating to the AzNetworking layer that the world was updated.
```

> 1. Sending an `ACK` to the server indicating the world was updated.

Author: I made a suggestion above to clarify this part, see line 33.

The roles offered for O3DE multiplayer hosts are:

* **Client** (`NetEntityRole::Client`): The lowest privilege role for a component. The smallest possible subset of network properties are replicated to this role, and its behavior is strictly read-only. Use this role when working with entities that should contain only presentation logic and act as a proxy for invoking RPCs. Examples of entities which should use this role are AIs and entities under the control of another host.

* **Autonomous** (`NetEntityRole::Autonomous`): A role with the _illusion_ of write access. Autonomous roles are usually assigned to components directly under local user control. These roles receive a larger amount of network information than a Client role. Autonomous roles can also take advantage of predictive networking.

* **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full write access and sees all network properties on the component.

> In addition to the previously described roles, O3DE has an additional role found only in multiserver instances:

Reviewer: Need a blank line here.
```suggestion

In addition to the previously described roles, O3DE has an additional role found only in multiserver instances:
```

* **Server** (`NetEntityRole::Server`): Provided to servers that *don't* have authority over a particular component. The proxy dummies out the networking properties, which the server should treat as if they were in the Client role.

## Related topics

| Topic | Description |
|--|--|
| [Networking](/docs/user-guide/networking/) | Learn about `AzNetworking`, the core networking framework that O3DE uses, and that the Multiplayer Gem is built on top of. |
| [Automate Source Generation from Templates with AzAutoGen](/docs/user-guide/engine/autogen/) | Learn about the AzAutoGen code generation system used to create [auto-components](./autocomponents) for projects using the Multiplayer Gem. |

> | [Open 3D Engine Multiplayer Gem API Reference](/docs/api/gems/multiplayer/) | The C++ API reference for classes that the Multiplayer Gem provides. |

Reviewer: Consider shortening
```suggestion
| [Multiplayer Gem API Reference](/docs/api/gems/multiplayer/) | The C++ API reference for classes that the O3DE Multiplayer Gem provides. |
```

The roles offered for O3DE multiplayer hosts are:

> * **Client** (`NetEntityRole::Client`): The lowest privilege role for a component. The smallest possible subset of network properties are replicated to this role, and its behavior is strictly read-only. Use this role when working with entities that should contain only presentation logic and act as a proxy for invoking RPCs. Examples of entities that should use this role are AIs and entities under the control of another host.

Reviewer 2: This may be inaccurate, please check with the networking team.

Reviewer 3: Generally accurate. The only change I would suggest is that you don't get to chose the role a networked entity is assigned, that's all handled automatically by the multiplayer gem itself. Hence I might change it to be more like this:
`* **Client** (`NetEntityRole::Client`): The lowest privilege role for a component. The smallest possible subset of network properties are replicated to this role, and its behavior is strictly read-only. Use this role when working with entities that should contain only presentation logic and act as a proxy for invoking RPCs. Examples of entities that should use this role are AIs and entities under the control of another host.`

Author: If that's the case, I'd go even further to specify that the Multiplayer Gem automatically assigns and handles the roles of networked entities at the top of this section. 

@kberg-amzn You omitted the example sentence in your suggested rewrite. I just wanted to verify whether that example still sounds accurate?

* **Autonomous** (`NetEntityRole::Autonomous`): A role with the _illusion_ of write access. Autonomous roles are usually assigned to components directly under local user control. These roles receive a larger amount of network information than a Client role. Autonomous roles can also take advantage of predictive networking.

> * **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full write access and sees all network properties on the component.

Reviewer 4: For this statement "It has full write access and sees all network properties on the component" -- does "sees" mean that it detects all network properties on the component?

Author: ```suggestion
* **Authority** (`NetEntityRole::Authority`): The role with ultimate authority. It has full read and write access to all network properties on the component.
```

In addition to the previously described roles, O3DE has an additional role found only in multiserver instances:

> * **Server** (`NetEntityRole::Server`): Provided to servers that *don't* have authority over a particular component. The proxy dummies out the networking properties, which the server should treat as if they were in the Client role.

Reviewer 3: Provided to servers that *don't* have authority over a particular entity. This role is strictly read-only, and all interaction with these entities should be handled using RPCs.
